### DISCUSSION TITLE
getting file transport to work

### DISCUSSION THREAD

**Discussion by mdagit** [adzejs/adze#179] [Q&A] [answered]
trying to follow https://adzejs.com/getting-started/capture-data.html#middleware

successfully did :
   npm install @adze/transport-file --save

nevertheless, i'm getting:

[vite] Internal server error: Failed to resolve entry for package "@adze/transport-file". The package may have incorrect main/module/exports specified in its package.json.

this is using both node and bun.  also it failed with both variants:

import {AdzeTransportFile} from '@adze/transport-file';
import AdzeTransportFile from '@adze/transport-file';

**Comment by AJStacy** [comment]
Hi @mdagit , I will investigate this issue, but could you provide me details about your project that you are installing the transport-file module into? Are you using commonjs or esm? Typescript? 

Thank you.

**Comment by mdagit** [comment]
My code is all pure typescript. Pure as the driven snow.

**Reply by AJStacy** [reply]
Could you paste your tsconfig here?

**Comment by mdagit** [comment]
Here is my tsconfig. After some digging, it seems this might be a vite issue, though honestly I can't make much sense of this:
   https://github.com/vitejs/vite/issues/1899
If i run my app directly with just "bun" (not with vite), then loading works fine. I only get this failure when running a vite dev server.

{
    "compilerOptions": {
        "target": "ESNext",
        "module": "ESNext",
        "moduleResolution": "Bundler",
        "strict": true,

        "skipLibCheck": true,

        "noImplicitAny": true,
        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "strictBindCallApply": true,
        "strictPropertyInitialization": true,
        "noImplicitThis": true,
        "alwaysStrict": true,

        "lib": [
            "ESNext",
            "dom.iterable"
        ],
        "types": [
            "@cloudflare/workers-types/2023-07-01",
            "vite/client"
        ],
        "jsx": "react-jsx",
        "jsxImportSource": "hono/jsx"
    },
    "exclude": [
        "node_modules",
        "unused-src"
    ]

}

**Reply by AJStacy** [reply]
It looks like your tsconfig is missing the values described in the setup page here:
https://adzejs.com/getting-started/installation.html#typescript-configuration

Try adding this configuration and let me know if that resolves it or not.

**Comment by mdagit** [comment]
You are right that I totally missed that instruction, but sadly even when I comply with that, vite still gives me unhelpful grief:

[plugin:vite:import-analysis] Failed to resolve entry for package "@adze/transport-file". The package may have incorrect main/module/exports specified in its package.json.

I haven't tried the magic incantations in that vite GitHub issue yet.

**Reply by mdagit** [reply]
Update: If my code does this:
import {AdzeTransportFile} from '@adze/transport-file/transport-file.js'
instead of:
import {AdzeTransportFile} from '@adze/transport-file'

then my current vite problems go away.

btw, the file @adze/transport-file/readme.md references the incorrect symbol TransportFile instead of AdzeTransportFile

**Reply by AJStacy** [reply]
The readme is correct since you can name default imports whatever you want. I'm going to spin up a vite/ts project to see if I can replicate the issue.

**Comment by AJStacy** [comment, marked as answer]
I have pushed a fix for this issue (v1.0.3)!
